### PR TITLE
Updated api template to use new ES5/6 features

### PR DESCRIPTION
**Pull request by svidhani**
### Requirements

NodeJs templates are still using the old way of writing code and now that there is a new way of writing code which is geared towards ES6 and more functional the template code is being updated. 

### Description of the Change

- Arrow functions
- use of const for readonly variable, function expression
- no need to write return keyword for single line return (objects)
- no need for checks for undefined
- use of object for lookup rather than if/else conditions
- more functional way of writing code
### Benefits

New style of writing code which is more geared towards ES6 in terms of syntax look and feel (concise).

### Possible Drawbacks

May not work with older version of node

### CODE REVIEW

File: templates/api-template-nodejs/deployment-env.yml [hunk 1]
```diff
@@ -1,3 +1,3 @@
-providerRuntime : nodejs4.3
+providerRuntime : nodejs6.1
```
[the remark this answers is not in the document]
Comment (devsatishm) [reply templates/api-template-nodejs/deployment-env.yml]: nodejs6.1 >> nodejs6.10 as per https://github.com/tmobile/jazz/blob/v1.8.1/builds/jenkins-build-pack-api/Jenkinsfile#L565

File: templates/api-template-nodejs/package.json [hunk 2]
```diff
@@ -9,8 +9,7 @@
     "test": "mocha"
   },
   "dependencies": {
-    "request": "*",
-    "deasync": "*"
+    "request": "*"
```
Comment (devsatishm) [inline templates/api-template-nodejs/package.json]: We wouldn't need it in the template. Let's remove.

Comment (svidhani) [reply templates/api-template-nodejs/package.json]: updated

File: templates/api-template-nodejs/deployment-env.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (svidhani) [reply templates/api-template-nodejs/deployment-env.yml]: updated

**Pull request merged**